packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -217,7 +339,7 @@ function AppLayoutSidebarNavItemInner({
 								background === 'body' ? 'backgroundShade' : 'backgroundShadeAlt'
 							],
 						color: boxPalette.foregroundText,
-						'& > span:first-of-type': packs.underline,
+						[`& > span:nth-of-type(${level})`]: packs.underline,
```

Comment:
I don't think this should use `level`? If we update the markup or add 3 levels of nav, this wouldn't selected the correct thing?

Reply:
We did update the markup and that's why the bug appeared in the first place. That's the trouble with generic selectors like this. Short of a rewrite, which I think is the best solution for the next feature-add, the only other thing I can think of is `level === 1 ? 1 : 2`.

Reply:
Oh, I see! The ndash is also in a span which makes this necessary 😬 This is why I like `data-` attributes for this stuff. Or use different elements, or perhaps select the last span (but I don't know what a badge does since we don't have any examples of that)

Reply:
Yeah, I don't mind `data-` attributes, used sparingly. They certainly solve the random bug issue caused by generic selectors. Ideally we lean into React.